File: 백준/Silver/2559. 수열/수열.js
const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
input = input.map((x) => x.trim("\r"));

const [N, K] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number);

let left = 0;
let right = 0;
let sum = 0;
let answer = Number.MIN_SAFE_INTEGER;

while (right <= N) {
  if (right-left === K) {
    answer = Math.max(answer, sum);
    sum -= numbers[left];
    left++;
  } else if (right-left < K) {
    sum += numbers[right];
    right++;
  } else if (right-left > K) {
    sum -= numbers[left];
    left++;
  }
}

console.log(answer);

File: 백준/Silver/2559. 수열/수열.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int n = Integer.parseInt(st.nextToken());
		int k = Integer.parseInt(st.nextToken());
		String[] nums = br.readLine().split(" ");

		int left = 0;
		int right = 0;
		int answer = Integer.MIN_VALUE;
		int sum = 0;

		while (right <= n) {
			if (right - left == k) {
				answer = Math.max(answer, sum);
				sum -= Integer.parseInt(nums[left++]);
			} else if (right - left > k) {
				sum -= Integer.parseInt(nums[left++]);
			} else if (right - left < k) {
				if (right >= n)
					break;
				sum += Integer.parseInt(nums[right++]);
			}
		}

		System.out.println(answer);

	}
}